platform/mds/mds-web/src/main/resources/webapp/js/directives.js

```diff
@@ -143,18 +143,33 @@
     * whether is selected for display in the jqGrid
     */
     function isSelectedField(name, selectedFields) {
-        var result = true;
-        if (selectedFields !== undefined && $.isArray(selectedFields)) {
-            $.each(selectedFields, function (i, sField) {
-                if(name === sField.basic.name) {
-                    result = true;
-                } else {
-                    result = false;
+        var i;
+        if (selectedFields) {
+            for (i = 0; i < selectedFields.length; i += 1) {
+                if (name === selectedFields[i].basic.name) {
+                    return true;
                 }
-                return (!result);
+            }
+        }
+        return false;
+    }
+
+    function handleGridPagination(pgButton, pager, scope, http) {
```

Comment:
Don't like the look of this function to be honest, it could use some comments, so that I have at least some clue what's happening here. Moreover, why not just save the page size in the retrieve records function of the controller? That way we can avoid an extra http call, and if we just do an update if the setting changed then this shouldn't affect performance much.